.github/workflows/nightly.yaml

```diff
@@ -0,0 +1,85 @@
+name: "nightly"
+run-name: "Nightly tests for ${{ github.ref_name }}"
+
+on:
+  schedule:
+    - cron: "22 0 * * *"  # Runs daily at 22:00 UTC
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  update-scanners:
+    runs-on: ubuntu-24.04
+    steps:
+
+      - name: Create github token
+        uses: actions/create-github-app-token@v1
+        id: app-token
+        with:
+          app-id: ${{ vars.ACTIONS_APP_ID }}
+          private-key: ${{ secrets.ACTIONS_APP_PRIVATE_KEY }}
+          owner: ${{ github.repository_owner }}
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ steps.app-token.outputs.token }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+
+      - name: Install dependencies
+        run: pip install requests
+
+      - name: Update scanner versions
+        run: python .github/scripts/update_scanners.py
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v7
+        id: pr
+        with:
+          title: Dependency update
+          branch: feature/deps-update
+          delete-branch: true
+          commit-message: ":arrow_up: Update scanner versions"
+          token: ${{ steps.app-token.outputs.token }}
```

Comment:
Any reason to use this action instead of a "simple" github script like we do in MetalK8s for example 

```yaml
      - name: Create pull request
        uses: actions/github-script@v7
        with:
          script: |
            const pr = await github.rest.pulls.create({
              owner: context.repo.owner,
              repo: context.repo.repo,
              head: "feature/merge-${{ env.RELEASE_TAG }}-tag",
              base: "development/${{ env.SHORT_VERSION }}",
              title: "Merge release tag for ${{ env.RELEASE_TAG }}"
            });
```